src/Components/Web/src/Virtualization/Virtualize.cs

```diff
@@ -213,15 +225,19 @@ protected override void BuildRenderTree(RenderTreeBuilder builder)
 
         _lastRenderedItemCount = 0;
 
-        // Render the loaded items.
-        if (_loadedItems != null && _itemTemplate != null)
+        if (!_loading && _itemCount == 0 && _emptyContent != null)
```

Comment:
Thanks for the update!

I just tried this out and it works great, but there's one last issue that could seem like a bug. I noticed that it renders the `EmptyContent` before the *first* load begins, so there is a very quick flash of that content, which is probably not what the developer would want. For example, in a Blazor Server prerendered app, when loading data asynchronously, the prerendered HTML would contain the "empty" content because it represents the state before loading starts.

Fortunately I think it's very easy to fix:

```suggestion
        if (_loadedItems != null && !_loading && _itemCount == 0 && _emptyContent != null)
```

... since `_loadedItems` becomes nonnull only after the first loading operation completes.

Do you think this will work?

Reply:
Didn't see that, thanks! :)